feat: add anvil ethereum devnet service to helm

aWN4Y25pa2EK opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added configurable local Ethereum devnet (Anvil) with optional persistence, health checks, resource limits, scheduling, and NodePort support.
  - Dynamic, data-driven Ingress path routing and added /api-docs route.
  - App and server environment vars exposed (VITE_SERVER_ENVIRONMENT, SERVER_ENVIRONMENT).
  - Service and PVC support for Anvil; secrets auto-point to Anvil RPC when enabled.
- Documentation
  - Comprehensive Anvil setup, usage, and troubleshooting guide.
- Chores
  - Bumped chart version to 0.8.1.
  - Updated local setup defaults for Ethereum RPC URL and Kind cluster image versions; added Kind CLI version check.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

openhands-agent committed:
feat: Add Anvil Ethereum devnet service to Helm charts

aWN4Y25pa2EK commented:
@Openhands deep review this PR in line with the initial specs and requirements

openhands-agent committed:
fix: correct YAML syntax errors in Anvil Helm templates

- Fix indentation in values.yaml server.image section (repository, tag, pullPolicy)
- Move nodePort field inside ports array in anvil-service.yaml
- Remove orphaned end tag in anvil-deployment.yaml
- Fix serviceAccountName reference to use arda-credit.fullname helper

These fixes resolve helm lint failures and ensure templates render correctly.

Co-authored-by: openhands <openhands@all-hands.dev>

aWN4Y25pa2EK commented:
@Openhands bump the Helm Chart version too

openhands-agent committed:
chore: bump Helm chart version to 0.7.0

aWN4Y25pa2EK commented:
@Openhands
-
Critical: The SERVER_ENVIRONMENT variable is missing from values-local.yaml.

Critical: The Anvil deployment's liveness and readiness probes use HTTP GET, which will likely fail as Anvil expects JSON-RPC POST requests. I recommend switching to tcpSocket probes.

Minor: Clarification is needed on how ETHEREUM_RPC_URL should be handled, especially with external secrets. I also suggested considering a longer state-interval for Anvil in local development.

openhands-agent committed:
fix: address Helm chart configuration issues

- Add missing SERVER_ENVIRONMENT variable to values-local.yaml
- Fix Anvil deployment probes to use tcpSocket instead of httpGet
  (Anvil expects JSON-RPC POST requests, not HTTP GET)
- Add clarifying comments for ETHEREUM_RPC_URL handling with external secrets
- Increase Anvil state-interval from 10s to 30s for better local dev performance

Co-authored-by: openhands <openhands@all-hands.dev>

aWN4Y25pa2EK committed:
feat(env): add Anvil-specific credentials to .env.example for local development
chore(helm): bump arda-credit chart version to 0.8.1 for new features
docs(README): document Anvil configuration options and usage for local development
fix(ingress): refactor ingress paths to use dynamic service names for better flexibility
fix(secrets): update ethereum-rpc-url handling to support Anvil service in secrets.yaml
chore(values): enable Anvil service by default in values.yaml for local/staging environments
chore(setup-cluster): adjust RPC_URL to use Anvil service URL for local development
chore(e2e): update local values to reference external secrets for Ethereum credentials

MSevey commented on .env.example:
```diff
@@ -36,6 +36,10 @@ ETHEREUM_CONTRACT_ADDRESS=
+# Anvil-specific credentials (used when Anvil is enabled)
+ANVIL_WALLET_PRIVATE_KEY=
+ANVIL_DEPLOYER_ADDRESS=
+
```
We shouldn't define new variables here. We should use the correct values for the existing variables based on the environment. 

Otherwise we have to add additional logic to switch between environment variables.

aWN4Y25pa2EK replied:
they have been removed!

aWN4Y25pa2EK committed:
chore(values.yaml): change server environment from production to staging for testing purposes

aWN4Y25pa2EK committed:
chore(env.example): remove Anvil-specific credentials to clean up example environment file and reduce confusion for users

aWN4Y25pa2EK committed:
fix(app-deployment.yaml): update environment variable name to include VITE_ prefix for frontend compatibility

aWN4Y25pa2EK committed:
chore(Makefile): add version check for Kind to ensure compatibility with required version
chore(kind-config.yaml): update Kind node image to v1.34.0 for improved stability and features
chore(setup-cluster.sh): remove unnecessary whitespace for cleaner script formatting
chore(values-local.yaml): update PostgreSQL image tag to 16.4.0 for latest features and security updates

aWN4Y25pa2EK committed:
chore(values-local.yaml): update PostgreSQL image tag from 16.4.0-debian-12-r14 to 15 for compatibility with local environment
